### PR TITLE
refactor: improve readability

### PR DESCRIPTION
**Pull request by IAKOBVS**

### CODE REVIEW

**Comment by IAKOBVS** [comment]
Are the values of the enums in PerlSymbolKind used externally? If not, we should remove the values or use numbers since a C compiler can optimize lookup of a switch statement with continuous values to be O(1).

**Review by bscan — APPROVED** [review]

**Comment by bscan** [comment]
> Are the values of the enums in PerlSymbolKind used externally? If not, we should remove the values or use numbers since a C compiler can optimize lookup of a switch statement with continuous values to be O(1).

Yes, these values are also specified in Inquisitor.pm. When the Navigator runs `perl -c` on an open file, it also runs the Inquisitor in the same process which inspects the symbol tables and prints out elements. It mostly looks for imported symbols like imported packages, subs, methods, and variables. You can see the format if you run this from the main directory. 

`perl -c -I server/src/perl -MInquisitor testWorkspace/mainTest.pl`

Essentially, `parser.ts` primarily parses the "current" file, while the Perl compiler is what parses all loaded dependencies.

File: server/src/completion.ts [hunk 1]
```diff
@@ -122,41 +115,48 @@ function getMatches(perlDoc: PerlDocument, symbol: string,  replace: Range): Com
...
+                    PerlSymbolKind.PathedField].includes(element.type)
+		    || !/^\$.*\-\>\w+$/.test(aligned))
+		    // FOO::BAR if Bar is a instance method or attribute (I assume them to be instance methods/attributes, not class)
+		     if ([PerlSymbolKind.LocalMethod,
+                        PerlSymbolKind.Method,
+			PerlSymbolKind.Field,
+			PerlSymbolKind.PathedField].includes(element.type)
+			|| /-:/.test(aligned) // We look things up like this, but don't let them slip through
+			|| /^\$.*::/.test(aligned)) // $Foo::Bar, I don't really hunt for these anyway
+		         return;
```
[the remark this answers is not in the document]
Comment (IAKOBVS) [reply server/src/completion.ts]: Changed it to multiple early returns.

File: server/src/completion.ts [hunk 2]
```diff
@@ -122,41 +115,48 @@ function getMatches(perlDoc: PerlDocument, symbol: string,  replace: Range): Com
     // const lcQualifiedSymbol = qualifiedSymbol.toLowerCase(); Case insensitive matches are hard since we restore what you originally matched on
 
     perlDoc.elems.forEach((elements: PerlElem[], elemName: string) => {
-        if(/^[\$\@\%].$/.test(elemName)) return; // Remove single character magic perl variables. Mostly clutter the list
+        if (/^[\$\@\%].$/.test(elemName))
+	    return; // Remove single character magic perl variables. Mostly clutter the list
```
[the remark this answers is not in the document]
Comment (IAKOBVS) [reply server/src/completion.ts:122]: #95 Can we use a code formatter? Currently, it's kind of a massive pain to deal with formatting.

**Comment by bscan** [comment]
Looks good to me, thanks again for the contributions! I'll merge some other stuff in today, and I'll try to format the files as I go.

**Comment by bscan** [comment]
I made a couple small touchups after the fact. There was an off-by-one issue in completion ($foo vs foo). Also, I continued adding to the subroutine signatures, and I needed to add some new code to close the call signatures when `)` is reached.  Thanks!

**Pull request merged**